backend/infrahub/message_bus/operations/event/branch.py

```diff
@@ -58,6 +75,16 @@ async def rebased(message: messages.EventBranchRebased, service: InfrahubService
             messages.TriggerIpamReconciliation(branch=message.branch, ipam_node_details=message.ipam_node_details),
         )
 
+    # for every diff that touches the rebased branch, recalculate it
+    component_registry = get_component_registry()
+    default_branch = registry.get_branch_from_registry()
+    diff_repository = await component_registry.get_component(DiffRepository, db=service.database, branch=default_branch)
+    diff_roots_to_refresh = await diff_repository.get_empty_roots(diff_branch_names=[message.branch])
+
```

Comment:
in the case of a rebase, we only need to recalculate diffs that include the branch that was rebased